src/Pux.purs

```diff
@@ -147,7 +148,7 @@ mapEffects a2b effmodel =
 -- | Wait for a specific event until returning the app state.
 waitEvent :: ∀ e ev st fx.
              (ev -> Boolean) -> App e ev st -> Aff fx st
-waitEvent until app = makeAff \error success -> waitEvent_ until app success
+waitEvent until app = makeAff \cb -> mempty <$ waitEvent_ until app (cb <<< pure)
```

Comment:
As an FYI, these functions leak channel subscriptions. There's no way to unsubscribe from a channel, and once the success branch is called, the subscriptions is still active even though the thread will no longer receive any updates.